draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -3050,6 +3050,62 @@ The constants used to compute y\_den are as follows:
 - k\_(4,1) = 0x1a0111ea397fe69a4b1ba7b6434bacd764774b84f38512bf6730d2a0f6b0f6241eabfffeb153ffffb9feffffffffa9d3 * I
 - k\_(4,2) = 0x12 + 0x1a0111ea397fe69a4b1ba7b6434bacd764774b84f38512bf6730d2a0f6b0f6241eabfffeb153ffffb9feffffffffaa99 * I
 
+# Cofactor Clearing {#clear-cofactor-suites}
+
+Some elliptic curve groups, e.g. G2 in BLS12-381, have a very large cofactor h\_eff (cf. {{cofactor-clearing}}).
+For these curves, we propose an alternative method for clearing the cofactor.
+
+## Cofactor Clearing on E' for BLS12-381 {#clear-cofactor-bls12381-g2}
+
+This fast cofactor clearing function was introduced by Budroni and Pintore {{BP18}}.
+It is based on Fuentes et al.'s method {{FKR11}}, and makes use of an efficiently-computable
+curve endomorphism `psi`.
+For the pairing-friendly curve BLS12-381, defined by the following constants {{draft-yonezawa-pfc-01}}:
+
+~~~
+p = 0x1a0111ea397fe69a4b1ba7b6434bacd764774b84f38512bf6730d2a0f6b0f6241eabfffeb153ffffb9feffffffffaaab
+r = 0x73eda753299d7d483339d80809a1d80553bda402fffe5bfeffffffff00000001
+x = -0xd201000000010000
+
+Zp = GF(p)
+Zp2.<i> = GF(p^2, modulus=[1, 0, 1])
+u = i+1
+
+E2 = EllipticCurve(Zp2, [0, 4*u])
+E1 = EllipticCurve(Zp, [0, 4])
+~~~
```

Comment:
This duplicates the definitions in `{{suites-bls12381-g2}}`. Instead, please refer to those definitions with a crossref, and then *only* define the constant you need (which I believe is `u = I + 1`, as `I` is defined in the appropriate Suites subsection).

I wouldn't bother with the `EllipticCurve()` stuff here. The goal should be to present it in a way that is readable to a human, not as syntactically correct Sage code (we could *also* add that if we wanted, but for the spec document itself that's beside the point).